### PR TITLE
Permissions

### PR DESCRIPTION
**Pull request by guibescos**
Update `init_mapping` to either accept the signature of the mapping account or the signature of the `master authority`. This is a blueprint to how the other instructions will get updated.

### CODE REVIEW

File: program/rust/src/utils.rs
Comment (jayantk) [inline program/rust/src/utils.rs:172]: you might have to fix this depending on how you merge the Rent PR

**Review by jayantk — COMMENTED** [review]
this makes sense to me, though suggest getting another approval to get another pair of eyes on the permission checking logic.

**Review by Reisen — APPROVED** [review]

**Pull request merged**
